File: VioletMarketplace/src/main/java/com/skilldistillery/marketplace/controllers/SearchController.java
package com.skilldistillery.marketplace.controllers;

import com.skilldistillery.marketplace.dtos.SearchResults;
import com.skilldistillery.marketplace.services.SearchServiceImpl;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("api")
@CrossOrigin({"*", "http://localhost:4301"})
public class SearchController {
    @Autowired
    private SearchServiceImpl searchService;

    @GetMapping("/search/{query}")
    public ResponseEntity<SearchResults> search(@PathVariable String query) {
        SearchResults sr = searchService.search(query);
        return ResponseEntity.ok(sr);
    }

}

File: VioletMarketplace/src/main/java/com/skilldistillery/marketplace/controllers/SearchController.kt
package com.skilldistillery.marketplace.controllers

import com.skilldistillery.marketplace.dtos.SearchResults
import com.skilldistillery.marketplace.services.SearchServiceImpl
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("api")
@CrossOrigin("*", "http://localhost:4301")
class SearchController(
    private val searchService: SearchServiceImpl,
) {
    @GetMapping("/search/{query}")
    fun search(@PathVariable query: String): ResponseEntity<SearchResults> {
        val sr = searchService.search(query)
        return ResponseEntity.ok(sr)
    }
}
